src/main/java/com/minecolonies/network/messages/BuildToolPlaceMessage.java

```diff
@@ -167,14 +166,26 @@ private static void handleHut(@NotNull World world, @NotNull EntityPlayer player
 
             @Nullable AbstractBuilding building = ColonyManager.getBuilding(world, buildPos);
 
-            if (building != null)
+            if (building == null)
             {
-                building.setStyle(style);
-                building.setRotation(rotation);
+                Log.getLogger().error("BuildTool: building is null!");
             }
             else
             {
-                Log.getLogger().error("BuildTool: building is null!");
+                if (building.getTileEntity() != null)
+                {
+                    final Colony colony = ColonyManager.getColony(world, buildPos);
+                    if (colony == null)
+                    {
+                        Log.getLogger().info("No colony for " + player.getName());
+                    }
+                    else
+                    {
+                        building.getTileEntity().setColony(colony);
+                    }
+                }
+                building.setStyle(style);
+                building.setRotation(rotation);
             }
         }
         else
```

Comment:
We could do the same with this else. For me it would make the code much more readable.